R/archive.R

```diff
@@ -878,17 +296,356 @@ epi_archive <-
 #'   time_type = "day",
 #'   other_keys = "county"
 #' )
-as_epi_archive <- function(x, geo_type, time_type, other_keys,
+#'
+new_epi_archive <- function(
+    x,
+    geo_type = NULL,
+    time_type = NULL,
+    other_keys = NULL,
+    additional_metadata = NULL,
+    compactify = NULL,
+    clobberable_versions_start = NA,
```

Comment:
suggest: eliminate `new_epi_archive`, keep it as `as_epi_archive`, or introduce a `new_` and `validate_` that are more minimal, and have `as_epi_archive` be the "helper" as described [here](https://adv-r.hadley.nz/s3.html).

suggest: either (a) make more defaults non-NULL, or (b) make `clobberable_versions_start` default also NULL (& replace with NA) if it's possible.  I'm not sure about (a) vs. (b).  (a) gives more type info, but also could hide other arg names in limited-width autocomplete windows.  With (a) approach, there can sometimes also be some issues with confusing defaults --- imagine we had `geo_type = guess_geo_type(x)` default and rebound `x` before `geo_type` was evaluated --- but it doesn't look like we rebind before assigning any of the defaults, so I don't _think_ we're in such a situation             .

Reply:
Alright, I added a validate_epi_archive and moved a bulk of the validation from new_epi_archive in there. Now as_epi_archive calls validate and then new. new_epi_archive is only used there, so this change should be fine for now, but in the future we can refactor some safe internal calls to as_epi_archive with new_epi_archive. There is still a bunch of validation and construction of the data.table object in new_epi_archive, so that's another thing that could be fixed.

I also made `clobberable_versions_start` default to NULL and then I set it to NA in that case. Just went with the consistent option for now, though we can easily switch it later.

Reply:
See #445